### PR TITLE
Medical - Add option to prevent scripted injuries on invulnerable units

### LINKED ISSUE #8593: ace_medical_fnc_addDamageToUnit not taking "isDamageAllowed" into account
**Is your enhancement related to a problem?**
As the title says, there is no check for invincibility in ace_medical_fnc_addDamageToUnit, meaning when you use it to apply ACE damage, you have to have check on your own end whether damage is allowed on the unit, which has to be local, making it more of a nuisance than necessary.

**Solution you'd like:**
A check for isDamageAllowed and the old ACE variable to happen within in the function, similar to #7505.

Maybe add an optional parameter to ace_medical_fnc_addDamageToUnit to check for vulernability, so it doesn't break or change current scripts, default value being off (i.e not to check for vulnerability).

### PR DESCRIPTION
**Pull request by mjc4wilton**
**When merged this pull request will:**
- Add optional parameter (default: true for backward compatibility) to enable adding damage to invulnerable units.
- Add exit condition check to `ace_medical_fnc_addDamageToUnit`
- Resolve #8593

### CODE REVIEW

File: addons/medical/functions/fnc_addDamageToUnit.sqf [hunk 1]
```diff
@@ -22,7 +23,7 @@
  */
 // #define DEBUG_TESTRESULTS
 
-params [["_unit", objNull, [objNull]], ["_damageToAdd", -1, [0]], ["_bodyPart", "", [""]], ["_typeOfDamage", "", [""]], ["_instigator", objNull, [objNull]], ["_damageSelectionArray", [], [[]]]];
+params [["_unit", objNull, [objNull]], ["_damageToAdd", -1, [0]], ["_bodyPart", "", [""]], ["_typeOfDamage", "", [""]], ["_instigator", objNull, [objNull]], ["_damageSelectionArray", [], [[]]], ["_overrideInvuln", true]];
```
Comment (BaerMitUmlaut) [inline addons/medical/functions/fnc_addDamageToUnit.sqf:26]: This line is getting fairly long and deserves reformatting. Additionally, I would add the third param entry (`[true]`).

**Review by BaerMitUmlaut — CHANGES_REQUESTED** [review]

**Comment by mjc4wilton** [comment]
Apologies for messy Git. VS Code has a stroke for some reason.

File: addons/medical/functions/fnc_addDamageToUnit.sqf [hunk 2]
```diff
@@ -22,8 +23,16 @@
...
+params [
+    ["_unit", objNull, [objNull]],
+    ["_damageToAdd", -1, [0]],
+    ["_bodyPart", "", [""]],
+    ["_typeOfDamage", "", [""]],
+    ["_instigator", objNull, [objNull]],
+    ["_damageSelectionArray", [], [[]]],
+    ["_overrideInvuln", true, [true]]
+];
+TRACE_6("addDamageToUnit",_unit,_damageToAdd,_bodyPart,_typeOfDamage,_instigator,_damageSelectionArray,_overrideInvuln);
```
Comment (BaerMitUmlaut) [inline addons/medical/functions/fnc_addDamageToUnit.sqf:35]:
```suggestion
TRACE_7("addDamageToUnit",_unit,_damageToAdd,_bodyPart,_typeOfDamage,_instigator,_damageSelectionArray,_overrideInvuln);
```

**Review by BaerMitUmlaut — CHANGES_REQUESTED** [review]

**Review by BaerMitUmlaut — APPROVED** [review]

File: addons/medical/functions/fnc_addDamageToUnit.sqf (hunk 2 above)
Comment (mjc4wilton) [reply addons/medical/functions/fnc_addDamageToUnit.sqf]: Why do you need to embarrass me like that?

File: addons/medical/functions/fnc_addDamageToUnit.sqf [hunk 3]
```diff
@@ -9,7 +9,8 @@
  * 2: Body part ("Head", "Body", "LeftArm", "RightArm", "LeftLeg", "RightLeg") <STRING>
  * 3: Projectile Type <STRING>
  * 4: Source <OBJECT>
- * 5: Non-directional damage source array (Optional) <ARRAY>
+ * 5: Non-directional damage source array <ARRAY> (default: [])
+ * 6: Override Invulnerability <BOOLEAN> (default: true)
```
Comment (jonpas) [inline addons/medical/functions/fnc_addDamageToUnit.sqf:13]:
```suggestion
 * 6: Override Invulnerability <BOO> (default: true)
```

File: addons/medical/functions/fnc_addDamageToUnit.sqf [hunk 4]
```diff
@@ -9,7 +9,8 @@
  * 2: Body part ("Head", "Body", "LeftArm", "RightArm", "LeftLeg", "RightLeg") <STRING>
  * 3: Projectile Type <STRING>
  * 4: Source <OBJECT>
- * 5: Non-directional damage source array (Optional) <ARRAY>
+ * 5: Non-directional damage source array <ARRAY> (default: [])
+ * 6: Override Invulnerability <BOO> (default: true)
```
Comment (jonpas) [inline addons/medical/functions/fnc_addDamageToUnit.sqf:13]:
```suggestion
 * 6: Override Invulnerability <BOOL> (default: true)
```

**Review by jonpas — APPROVED** [review]

File: addons/medical/functions/fnc_addDamageToUnit.sqf (hunk 3 above)
Comment (mjc4wilton) [reply addons/medical/functions/fnc_addDamageToUnit.sqf]: Halloween spirit is still strong I see

**Pull request merged**